app/components/TransactionHistory/TransactionHistoryPanel/TransactionHistoryPanel.jsx

```diff
@@ -24,11 +30,39 @@ export default class TransactionHistory extends React.Component<Props> {
         <Transactions
           className={styles.transactions}
           transactions={transactions}
+          pendingTransactions={pendingTransactions || []}
         />
       </Panel>
     )
   }
 
+  async pruneReturnedTransactionsFromStorage() {
+    const {
+      transactions,
+      pendingTransactions,
+      address,
+      net,
+      getPendingTransactionInfo,
+    } = this.props
+    if (
+      transactions &&
+      transactions.length &&
+      pendingTransactions &&
+      pendingTransactions.length
+    ) {
+      const confirmed = transactions.map(tx => tx.txid)
+      // NOTE: removes the '0x' prepended to every txId
+      const pending = pendingTransactions.map(tx => tx.txid.substring(2))
+      const toBePurged = intersection(confirmed, pending)
+      // eslint-disable-next-line
+      for (const id of toBePurged) {
+        // eslint-disable-next-line
+        await pruneConfirmedOrStaleTransaction(address, id)
+        getPendingTransactionInfo({ address, net })
```

Comment:
I haven't dug into the code but something here feels like a perf bottleneck.
1. Multiple sequential calls to `getPendingTransactions` and `getPendingTransactionInfo`.
2. The `await` makes these run in sequence . Can instead run in parallel?

Reply:
So I am running these in sequence on purpose in case one of the transaction ids stored locally invalid this allows me to sequentially filter them out. Maybe there is a way to do this in parallel that I am not thinking of?

Reply:
Understood. I see also that it updates a single localStorage key which should not be updated in parallel. I'll let you know if I find anything in particular.

Reply:
The part I'm having trouble understanding is the execution of `getPendingTransactionInfo` here. It doesn't expect the return value and it's costly (creates RPC client, storage access, tx parsing, etc.). Can you explain?

Reply:
This is an excellent point (I just addressed this issue in my most recent commit) basically the intention was to update pendingTransactions via dispatching an action... Like you said its much more performant to just synchronously filter out the confirmed transactions (abstracts returned from neoscan API) and not dispatch the action 👍